internal/cluster/cluster.go

```diff
@@ -668,6 +674,70 @@ func (m *Manager) UpdateLogsProgress(ctx context.Context, c *common.Cluster, pro
 	return err
 }
 
+func (m *Manager) UpdateInstallProgress(ctx context.Context, clusterID strfmt.UUID) error {
+	log := logutil.FromContext(ctx, m.log)
+
+	cluster, err := common.GetClusterFromDB(m.db, clusterID, common.UseEagerLoading)
+	if err != nil {
+		log.WithError(err).Error("Failed to get cluster from DB")
+		return err
+	}
+
+	var totalHostsDoneStages, totalHostsStages float64
+	for _, h := range cluster.Hosts {
+		stages := m.hostAPI.GetStagesByRole(h.Role, h.Bootstrap)
+		currentIndex := m.hostAPI.IndexOfStage(h.Progress.CurrentStage, stages)
+		totalHostsDoneStages += float64(currentIndex + 1)
+		totalHostsStages += float64(len(stages))
+	}
+
+	installingStagePercentage := int64((totalHostsDoneStages / totalHostsStages) * 100)
+	totalPercentage := int64(common.ProgressWeightPreparingForInstallationStage*float64(cluster.Progress.PreparingForInstallationStagePercentage) +
```

Comment:
If we are in installation-in-progress shouldn't the preparing-for-installation percentage currently be complete?

Reply:
It is complete, but it was already updated at an earlier stage therefore we just need to update the new part (the other fields of the embedded struct in DB will remain unchanged.

Reply:
If you mean why I use the old data from DB instead of just writing `100` instead it is just because it felt better coding for me (if anyhow something didn't reach 100% we should reflect it in `totalPercentage` so we can fix it - otherwise we won't see the issue.